addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-connection-service.test.js

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const ServicesContainer = require('@aws-ee/base-services-container/lib/services-container');
+const JsonSchemaValidationService = require('@aws-ee/base-services/lib/json-schema-validation-service');
+const Logger = require('@aws-ee/base-services/lib/logger/logger-service');
+
+// Mocked dependencies
+jest.mock('@aws-ee/base-api-services/lib/jwt-service');
+const JwtService = require('@aws-ee/base-api-services/lib/jwt-service');
+
+jest.mock('@aws-ee/key-pair-mgmt-services/lib/key-pair/key-pair-service');
+const KeyPairServiceMock = require('@aws-ee/key-pair-mgmt-services/lib/key-pair/key-pair-service');
+
+jest.mock('@aws-ee/base-services/lib/settings/env-settings-service');
+const SettingsServiceMock = require('@aws-ee/base-services/lib/settings/env-settings-service');
+
+jest.mock('@aws-ee/base-services/lib/audit/audit-writer-service');
+const AuditServiceMock = require('@aws-ee/base-services/lib/audit/audit-writer-service');
+
+jest.mock('@aws-ee/base-services/lib/plugin-registry/plugin-registry-service');
+const PluginRegistryServiceMock = require('@aws-ee/base-services/lib/plugin-registry/plugin-registry-service');
+
+jest.mock('../../environment-dns-service.js');
+const EnvironmentDnsServiceMock = require('../../environment-dns-service.js');
+
+jest.mock('../environment-sc-service');
+const EnvironmentSCServiceMock = require('../environment-sc-service');
+
+jest.mock('../environment-sc-keypair-service');
+const EnvironmentScKeyPairServiceMock = require('../environment-sc-keypair-service');
+
+const EnvironmentScConnectionService = require('../environment-sc-connection-service');
+
+describe('EnvironmentScConnectionService', () => {
```

Comment:
I was expecting to see an assert where we check the RStudioUrl after mocking a bunch of calls. Do we not care about what the final RStudio URL looks like?

Reply:
I was considering that to be tested manually with an actual certified domain name, but adding a unit test to make sure the URL structure stays as expected makes sense. I'll add a test for that.